6_nodejs/week3/app.js

```diff
@@ -0,0 +1,50 @@
+const knex = require("knex")({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "0.0.0.0",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "pass",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true, 
+  },
+});
+
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knex.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy.length > 0) {
+        //query = query.orderByRaw(orderBy); the following request will drop the contacts table /api/contacts?sort=DROP TABLE contacts
+      query = query.orderByRaw(knex.raw('??', [orderBy]));
+    }
+  }
+
+
+  console.log("SQL", query.toSQL().sql);
+
+  try {
+    const data = await query;
```

Comment:
Shouldn't this be `query()`?

Reply:
Not sure, because is calling a query _knex.select("*").from("contacts")_ in that case. Anyway, that part of the code was wrote before, so im not 100% sure if I modify that I need to rewrite te rest of the code, if you guide me to do it would be great to don't break anything else 😊